14th sprint was added: (#8)

AntonSik opened the pull request:
- media player and playing tracks
- debounce touching on recyclerView's items
- loading progress bar
- search tracks without button

AntonSik committed:
14th sprint was added:
- media player and playing tracks
- debounce touching on recyclerView's items
- loading progress bar
- search tracks without button

AntonSik committed:
14th sprint was added: (#8)

- media player and playing tracks
- debounce touching on recyclerView's items
- loading progress bar
- search tracks without button

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -93,5 +125,74 @@ class AudioPlayerActivity : AppCompatActivity() {
+            override fun run() {
+
+                val elapsedTime = System.currentTimeMillis() - startTime
+                val elseTime = duration - elapsedTime
```
Если делать таймер таким образом, то он быстро рассинхронизируется с реальным плеером. У плеера есть функция `getCurrentPosition`, можно использовать её

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -15,6 +18,14 @@ import java.text.SimpleDateFormat
+        private const val STATE_PLAYING = 2
+        private const val STATE_PAUSED = 3
+        private const val UPDATING_DELAY = 400L
+        private const val TRACK_DURATION = 30000L
```
Не стоит это хардкодить, т.к. в любой момент сервер может поменять это значение и присылать уже не по 30с отрывки, а по 40, например. И приложение тогда будет работать некорректно

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -38,6 +43,11 @@ class SearchActivity : AppCompatActivity(), OnClickListenerItem {
     private val itunesService = retrofit.create(ItunesApi::class.java)
 
     private var savedText: String? = null
+    private var searchRunnable = Runnable { tracksSearch() }
```
Этот раннабл лучше удалять в методе `onDestroy` из очереди аналогично тому, как ты удаляешь в `AudioPlayerActivity`

ArturNurtdinov commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -74,6 +73,15 @@
+                android:id="@+id/progressBar"
+                android:layout_width="44dp"
+                android:layout_height="44dp"
+                android:layout_marginTop="140dp"
```
Все размеры лучше вынести в dimens, особенно повторяющиеся

ArturNurtdinov reviewed (commented):
1. Заголовок "Вы искали" отображается при обычном поиске треков:
![image](https://github.com/AntonSik/PlaylistMaker/assets/49075521/a980d4e6-a6ff-417b-975b-597e569584b0)

AntonSik committed:
14th sprint was added:
- media player and playing tracks
- debounce touching on recyclerView's items
- loading progress bar
- search tracks without button

AntonSik committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt
#	app/src/main/java/com/example/playlistmaker/SearchActivity.kt
#	app/src/main/res/layout/activity_search.xml

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -168,24 +167,18 @@ class AudioPlayerActivity : AppCompatActivity() {
-                val elapsedTime = System.currentTimeMillis() - startTime
-                val elseTime = duration - elapsedTime
+                val countedTime = startTime + mediaPlayer.currentPosition
 
-                if (elseTime > 0) {
+                if (countedTime > 0) {
```
А зачем вообще нужно `countedTime`, если в `timePlaying.text` всё равно передаётся просто `mediaPlayer.currentPosition`? У тебя вроде как всегда `countedTime` будет больше 0, т.к. ты передаёшь туда `System.currentTimeMillis()`, а это всегда большое положительное число
